### ISSUE TITLE
Very slow inference on 3080

### ISSUE THREAD

**Issue by jbrownkramer** [apple/ml-depth-pro#64]
I would expect the 3080 to be, say, 30 to 50% slower than a V100.  But  it takes 45 seconds to run inference.  The GPU utilization is very low (10% or so).  Any idea how this could be fixed?

**Comment by jbrownkramer** [comment]
I did model = model.cuda()  and image = image.cuda() and I got higher GPU usage and more like 5 to 8 seconds of inference time.

Ran it again (without reloading the model or the image).  Massive RAM usage and much longer inference time.

**Comment by jbrownkramer** [comment]
Setting precision=torch.float16 seems to fix all of this.  The initial inference takes about 1.5s, and subsequent inferences take about .6s.

You could consider updating the model loading part in the python code snippet to

```
# Load model and preprocessing transform
model, transform = depth_pro.create_model_and_transforms(device=torch.device("cuda"),precision=torch.float16)
model.eval()
```

**Comment by lattebyte** [comment]
hi mate, i used 3090 which took approx below 1s for inference (~300ms,  high GPU usage, low vram and low ram unless there is bug i got initially), video input from usb camera rather than local image, if interested, code is here https://github.com/lattebyte/DepthPro-Windows-GPU

**Comment by Itachi-6** [comment]
@jbrownkramer I also have the same problem of larger inference time for a single image. It takes around 30-40 sec to process 1 image. I used the input you gave but it is giving me OutOfMemory Error. Could you please help me on what to do now?

**Comment by jbrownkramer** [comment]
I don't have a  GTX-1650 at my disposal.  You could try setting the precision even lower.

**Comment by jbrownkramer** [comment]
> Hi @jbrownkramer , I tried your suggestion on a 3090.
> 
> If set `float16`, it takes ~17s; if `default,` it tasks ~18s.
> 
> And it seems they already set it to Half? #17
> 
> Do you have any idea about this? Thanks!

@csyhping in issue #17 the user is running through run.py, which calls create_model_and_transforms on gpu if you have it, and half precision.  However if you look at the definition of create_model_and_transforms

```
def create_model_and_transforms(
    config: DepthProConfig = DEFAULT_MONODEPTH_CONFIG_DICT,
    device: torch.device = torch.device("cpu"),
    precision: torch.dtype = torch.float32,
) -> Tuple[DepthPro, Compose]:
```

You see that by default it is float32 and running on cpu.  I guess make sure you're doing

```
model, transform = depth_pro.create_model_and_transforms(device=torch.device("cuda"),precision=torch.float16)
model.eval()
```

And verify that your GPU is being well utilized when you run inference.

**Comment by xulsup** [comment]
@jbrownkramer 

Thanks for your help.
```
model, transform = depth_pro.create_model_and_transforms(device=torch.device("cuda:0"),precision=torch.float16)
model.eval()
```

On my laptop with a 4070 graphics card. Its reasoning speed is about half faster than the CPU. The CPU reasoning requires about 30 seconds, and 4070 is about 15 seconds.

**Comment by xulsup** [comment]
@Itachi-6  
What he meant was to let the model actually perform reasoning on a simple example, and then feed the model with the samples you really need to reason about. From this difference, the speed of subsequent reasoning will be faster.

**Comment by AlessandroScherl** [comment]
5090 and float16 precision got me this:

=== Timing Results ===
Model load time: 8.248 seconds
Warmup inference: 0.315 seconds
Average inference time: 0.127 seconds

**Comment by leo038** [comment]
我在RTX 3060上， 输入大小1280*720, 推理时间大约2.5。  作者说的0.3s 2.25M 像素是在什么GPU上呀?
